lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -164,6 +168,15 @@ export default function ContentSelector({
     onSelectContent({ type: 'file', file });
   };
 
+  const selectCanvasPage = (page: File) => {
```

Comment:
This argument has the type `File` although the resource is actually a page. This could be handled by:

1. Adding a separate `onSelectPage` callback to `LMSFilePicker`
2. Making the `onSelectFile` argument `File | Page` and then using a cast to convert it to the expected type
3. Making `LMSFilePicker` generic over the type of resource and then `onSelectFile` would emit the corresponding kind of resource.

Of these (2) is probably the easiest.

Reply:
> I'm thinking making it generic (like ResourcesList) probably makes sense, and would allow to conditionally define all the user-facing dynamic texts, without having to define a prop for each one of them.

How will the conditional definitions work? Maybe the easiest way to show what you mean will be to sketch some code and show me.

Reply:
Let me try to sketch something and push it here.